google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/GoogleCloudApiSelectorPanel.java

```diff
@@ -269,27 +274,47 @@ private void updateManagementUI() {
   }
 
   /**
-   * Populates the BOM {@link JComboBox} with the fetched versions. If there are no versions
-   * returned, then the BOM UX is hidden.
+   * Populates the BOM {@link JComboBox} with the fetched versions. If there is already a
+   * preconfigured BOM in the corresponding module's pom.xml, then its version will be preselected.
+   * If there are no versions returned and there is no preconfigured BOM, then the BOM UX is hidden.
    *
    * <p>Sorts the displayable versions in reverse order, and limits the number shown to some value
    * N.
    */
   // TODO (eshaul): make async with loader icons
   private void populateBomVersions() {
-    List<Version> bomVersions =
+    List<String> availableBomVersions =
         Lists.newArrayList(CloudApiMavenService.getInstance().getBomVersions());
 
-    if (bomVersions.isEmpty()) {
+    Optional<String> configuredBomVersion =
+        CloudLibraryProjectState.getInstance(project)
+            .getCloudLibraryBom(modulesComboBox.getSelectedModule())
+            .map(MavenDomDependency::getVersion)
```

Comment:
This sequence would work great with all `null` probabilities and will just return empty optional if something is missing, right?

Reply:
correct; with the update in this PR to `getCloudLibraryBom` to return a default value, this sequence will either return an optional with a configured BOM version, or an empty optional.

Reply:
That's neat, great case for mapping Optionals!